webapp/tests/karma/ts/modules/contacts/contacts-content.component.spec.ts

```diff
@@ -240,4 +294,180 @@ describe('Contacts content component', () => {
+      expect(globalActions.setRightActionBar.args[0][0].canDelete).to.equal(true);
+      expect(globalActions.setRightActionBar.args[0][0].canEdit).to.equal(true);
+    }));
+
+    it('should disable edit when user is not admin and facility is home place ', fakeAsync(() => {
+      sinon.resetHistory();
+      sessionService.isAdmin.returns(false);
+      userSettingsService.get.resolves({ facility_id: 'district-123' });
+
+      component.ngOnInit();
+      flush();
+
+      expect(globalActions.setRightActionBar.callCount).to.equal(1);
+      expect(globalActions.setRightActionBar.args[0][0].canDelete).to.equal(false);
+      expect(globalActions.setRightActionBar.args[0][0].canEdit).to.equal(false);
+    }));
+
+    it('should enable edit when user is not admin and facility is not home place ', fakeAsync(() => {
+      sinon.resetHistory();
+      sessionService.isAdmin.returns(false);
+      component.userSettings = { facility_id: 'district-9' };
+
+      component.ngOnInit();
+      flush();
+
+      expect(globalActions.setRightActionBar.callCount).to.equal(1);
+      expect(globalActions.setRightActionBar.args[0][0].canDelete).to.equal(false);
+      expect(globalActions.setRightActionBar.args[0][0].canEdit).to.equal(true);
+    }));
+
+    it('should filter contact types to allowed ones from all contact forms', fakeAsync(() => {
+      sinon.resetHistory();
+      contactTypesService.getChildren.resolves([
+        {
+          id: 'type1',
+          create_form: 'form:contact:create:type1',
+        },
+        {
+          id: 'type2',
+          create_form: 'form:contact:create:type2',
+        },
+        {
+          id: 'type3',
+          create_form: 'form:contact:create:type3',
+        },
+      ]);
+      const forms = [
+        { _id: 'form:contact:create:type3' },
+        { _id: 'form:contact:create:type2' },
+      ];
+
+      component.ngOnInit();
+      flush();
+
+      expect(xmlFormsService.subscribe.callCount).to.equal(2);
+      expect(xmlFormsService.subscribe.args[0][0]).to.equal('ContactForms');
+      expect(xmlFormsService.subscribe.args[0][1]).to.deep.equal({ contactForms: true });
+
+      xmlFormsService.subscribe.args[0][2](null, forms);
+
+      expect(globalActions.updateRightActionBar.callCount).to.equal(1);
+      expect(globalActions.updateRightActionBar.args[0][0]).to.deep.equal({
+        childTypes: [
+          {
+            menu_icon: 'fa-building',
+            menu_key: 'Add place',
+            permission: 'can_create_places',
+            types: [
+              {
+                create_form: 'form:contact:create:type2',
+                id: 'type2'
+              },
+              {
+                create_form: 'form:contact:create:type3',
+                id: 'type3'
+              }
+            ]
+          }
+        ]
+      });
+    }));
+
+    it('should filter contact types to allowed ones from selected contact forms', fakeAsync(() => {
```

Comment:
I think this test should be renamed, as it's not about contact forms, it's about report forms. 

Also the contents of the test is around Contact forms, when it should be around report forms. Children are not relevant here, are they not? Just the selected contact is relevant.

Reply:
Oh yes, Fixed!